### PR TITLE
fix: jsonld credential format identifier version

### PR DESCRIPTION
**Pull request by NB-MikeRichardson**
1. Fix the JSONLD_VC format string (it was missing a "v" character).

2. Remove the check credential proofType === request proofType. I believe this check is unnecessary (and always fails when testing with aca-py)

### CODE REVIEW

**Review by TimoGlastra — CHANGES_REQUESTED** [review]
the v prefix changes makes sense, but I think we need to keep the proof type check

File: packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormatService.ts [hunk 1]
```diff
@@ -356,10 +356,6 @@ export class JsonLdCredentialFormatService implements CredentialFormatService<Js
       )
     }
 
-    if (credential.proof.type !== request.options.proofType) {
-      throw new AriesFrameworkError('Received credential proof type does not match proof type from credential request')
-    }
```
[the remark this answers is not in the document]
Comment (NB-MikeRichardson) [reply packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormatService.ts:361]: We are getting "assertionMethod" in the credential but the request proof type is undefined. I can't see anywhere in our code were we set proof type for an outgoing credential request, hence why I was thinking the test is superfluous. Unless I've missed something

[the remark this answers is not in the document]
Comment (NB-MikeRichardson) [reply packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormatService.ts:361]: The other possibility is that the incoming test data is wrong, and credential proof type should not be set which would imply a problem somewhere else than AFJ code.

[the remark this answers is not in the document]
Comment (NB-MikeRichardson) [reply packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormatService.ts:361]: I'll do some investigation and see what I can find.

[the remark this answers is not in the document]
Comment (NB-MikeRichardson) [reply packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormatService.ts:361]: OK so I discovered we copy proof type and proof puprose from the offer to the request so I guess that means there is something wrong on the incoming offer message that needs investigating.

I've put the check back in. This PR can now be merged with just the one fix.

**Review by TimoGlastra — APPROVED** [review]

**Pull request merged**